public/app/features/canvas/element.ts

```diff
@@ -27,6 +27,29 @@ export interface CanvasElementOptions<TConfig = any> {
   placement?: Placement;
   background?: BackgroundConfig;
   border?: LineConfig;
+  connections?: CanvasConnection[];
+}
+
+// Unit is percentage from the middle of the element
+// 0, 0 middle; -1, -1 bottom left; 1, 1 top right
+export interface ConnectionCoordinates {
+  x: number;
+  y: number;
+}
+
+export enum ConnectionPath {
+  Straight = 'straight',
+}
+
+export interface CanvasConnection {
+  source: ConnectionCoordinates;
+  target: ConnectionCoordinates;
+  sourceName?: string;
+  targetName?: string;
+  color: string;
```

Comment:
since color+size are hardcoded in this PR anyway, can we remove them for now?

They should likely use the same config as elements -- *perhaps* with added logic to match the elements color config?